spec/models/sms/sms_decoder_spec.rb

```diff
@@ -531,6 +531,83 @@
     end
   end
 
+  describe "nested groups" do
+    let(:form) do
+      create_form(
+        questions: %w[integer multilevel_select_one_as_text_for_sms],
+        default_option_names: true
+      )
+    end
+
+    # QingGroup (root)
+    #   Questioning
+    #   Questioning (multilevel)
+    #   QingGroup
+    #     Questioning
+    #     Questioning
+    # TODO: add another nested group
+    #     QingGroup
+    #       Questioning
+    before do
+      qing_group = create_form(questions: %w[integer integer]).root_group
+      qing_group.parent = form.root_group
+      qing_group.rank = 3
+      qing_group.save!
+    end
```

Comment:
Please use the factory to make the doubly nested group. Search form_odk_rendering_spec.rb for `nested repeat group form` for an example.